src/engine/strat_engine/blockdev.rs

```diff
@@ -225,7 +241,7 @@ impl BlockDev {
         }
     }
 
-    pub fn wipe_metadata(&mut self) -> EngineResult<()> {
+    pub fn wipe_metadata(&self) -> EngineResult<()> {
```

Comment:
```Pool::destroy()``` consumes self, not a reference to self, so I think you can do a whole lot in the body of the method.